### PR TITLE
chore: adaptations for nightly-2024-08-03

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Hamiltonian.lean [hunk 1]
```diff
@@ -79,28 +79,27 @@ lemma IsHamiltonianCycle.map {H : SimpleGraph β} (f : G →g H) (hf : Bijective
...
     · cases hp.ne_nil rfl
-    simp only [support_cons, List.count_cons, List.map_cons, List.head_cons, beq_iff_eq,
-      hf.injective.eq_iff, add_tsub_cancel_right]
+    simp only [support_cons, List.count_cons, beq_iff_eq, List.head_cons, hf.injective.eq_iff,
+      add_tsub_cancel_right]
     exact hp.isHamiltonian_tail _
 
 lemma isHamiltonianCycle_isCycle_and_isHamiltonian_tail  :
-    p.IsHamiltonianCycle ↔ p.IsCycle ∧ p.tail.IsHamiltonian :=
+    p.IsHamiltonianCycle ↔ ∃ h : p.IsCycle, (p.tail h.not_nil).IsHamiltonian :=
```
Comment (kbuzzard) [inline Mathlib/Combinatorics/SimpleGraph/Hamiltonian.lean:92]: Is the name of this lemma still OK?

**Review by kbuzzard — APPROVED** [review]
Thanks! Note that CI is failing.

bors d+

**Comment by semorrison** [comment]
It wasn't *really* failing: it was a transient network failure retrieving the cache.

File: Mathlib/Combinatorics/SimpleGraph/Hamiltonian.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Combinatorics/SimpleGraph/Hamiltonian.lean:92]: I guess so?

**Pull request merged**